extension.js

```diff
@@ -112,9 +113,49 @@ function description(app) {
     return appName + ' → ' + app.get_title();
 }
 
-function updateHighlight(boxes) {
-  boxes.forEach(box => box.whole.remove_style_class_name('switcher-highlight'));
-  boxes.length > cursor && boxes[cursor].whole.add_style_class_name('switcher-highlight');
+function updateHighlight(boxes, query) {
+  boxes.forEach(box => {
+    // Remove previous selection highlight
+    box.whole.remove_style_class_name('switcher-highlight');
+
+    // Don't highlight description if there's no input
+    if (query == "")
+      return;
+
+    // Identify substring parts to be highlighted
```

Comment:
maybe everything from this down could be a separate function returning what is `result` below so that here it would only say `const result = whatever`.